src/plugins/controls/public/options_list/components/options_list_popover_suggestions.tsx

```diff
@@ -191,12 +198,19 @@ export const OptionsListPopoverSuggestions = ({
           )}
           emptyMessage={<OptionsListPopoverEmptyMessage showOnlySelected={showOnlySelected} />}
           onChange={(newSuggestions, _, changedOption) => {
-            const key = changedOption.key ?? changedOption.label;
+            if (!fieldSpec || !changedOption.key) {
```

Comment:
Same here, callbacks can not throw because then the error is bubbled up to error boundary or top level react component

Reply:
See https://github.com/elastic/kibana/pull/188789#discussion_r1693507540 - I'm not convinced it's worth handling here, since this error should never actually be seen 🤷 I could technically just return early on that case rather than throwing an error - perhaps that is clearer?

Reply:
Switched to early return in c9672eedee5c99962f734984936a1a9da0001d6a

Reply:
How about a console.warn? That way if this ever occurs, there is still some log that we could use to debug the problem?

Reply:
Good call - added `console.warn` in 0edfa0fa1a0d5a85cae025d1d711ec17e523f7e5 :+1: